src/Commands/WaContent.php

```diff
@@ -125,6 +128,82 @@ private function createPost($waContent)
+        $translationPostIds = $postTranslations->map(
+            function ($translation_id, $locale) use ($waContent) {
+                if (null == (env($endpoint = strtoupper('WHITEALBUM_ENDPOINT_'.$locale)))) {
+                    WP_CLI::warning(sprintf('%s has not been defined in your ENV file.', $endpoint));
+                    return;
+                }
+                $repository = new ContentRepository(env($endpoint));
+                $translation_post_id = null;
+                if ($translation_post_id = WpComposite::id_from_white_album_id($translation_id)) {
+                    WP_CLI::success(sprintf('found imported translation: %s', $translation_post_id));
+                } else {
+                    //if associated post isn't imported yet, check both articles and galleries
+                    $translation = collect(
+                        [
+                        'article' => ContentRepository::ARTICLE_RESOURCE,
+                        'gallery' => ContentRepository::GALLERY_RESOURCE,
+                        ]
+                    )->map(
+                        function ($endpoint, $contentType) use ($translation_id,$repository, $locale) {
+                            WP_CLI::line(sprintf('looking for a %s with the id: %s on %s ', $contentType, $translation_id, $locale));
+
+                            if ($translation  = $repository->find_by_id($translation_id, $endpoint)) {
+                                WP_CLI::line(sprintf('found translation: %s', $translation->widget_content->title));
+                                $this->importComposite($translation);
+                                $translationPostId = WpComposite::id_from_white_album_id($translation_id);
+                                return $translationPostId;
+                            }
+                            WP_CLI::line(sprintf('no translations for %s of type %s found.', $locale, $contentType));
+                        }
+                    )->reject(
+                        function ($item) {
+                            return empty($item);
+                        }
+                    );
+                    if (empty($translation)) {
+                        WP_CLI::line(sprintf('no translations for %s found.', $translation_id));
+                        return;
+                    }
+
+                    return $translation->first();
+                }
+                WP_CLI::line(sprintf('found post id: %s', $translation_post_id));
+                return $translation_post_id;
+            }
+        )->reject(
+            function ($item) {
+                return empty($item);
+            }
+        );
+        if ($translationPostIds->isEmpty()) {
+            WP_CLI::error('locale specific endpoints are missing', true);
+            return;
+        }
+
+        pll_save_post_translations($translationPostIds);
```

Comment:
There should hopefully be a LanguageProvider wrapper for this, instead of using pll directly.